distributed/comm/ucx.py

```diff
@@ -89,7 +89,7 @@ def init_once():
         return
 
     # remove/process dask.ucx flags for valid ucx options
-    ucx_config = _scrub_ucx_config()
+    ucx_config, ucx_environment = _prepare_ucx_config()
 
     # We ensure the CUDA context is created before initializing UCX. This can't
     # be safely handled externally because communications in Dask start before
```

Comment:
I note here that this check is not complete since it will only check the high level options for TLS. Shall I extend to also look in `ucx_environment` and `os.environ`?

Reply:
Yes, I think it would be better. Since we're opening the door for users to set arbitrarily any variables now, it may be wise to do so. By the way, it tests for `"cuda_copy" in ucx_config["TLS"]`, but it's also allowed to set `"cuda"` (which would include not only `cuda_copy`, but also `cuda_ipc`), could you change the check for `"cuda" in ucx_config["TLS"]` as well?

Reply:
Is that right? If UCX has cuda support, then `UCX_TLS=all` will need the same handling no? I'll do best effort specific checking.

Reply:
> If UCX has cuda support, then UCX_TLS=all will need the same handling no?

Yeah, but the problem is with `UCX_TLS=all` we don't know what really is available, nor if the user will indeed require CUDA. For example, we don't want users of `LocalCluster` (the CPU one) to have a CUDA context being created just because it can. The solution for that, although not optimal, was to allow creating the context if `distributed.comm.ucx.create-cuda-context=True`.

Reply:
Also this requirement (of having a CUDA context created before instantiating UCX to ensure correct CUDA<->IB mappings) is supposed to go away with UCX protocol v2, which is currently under development. After that, we will hopefully be able to ditch context creation here.

Reply:
Added best-effort checking.